week4/src/Components/BeerList.jsx

```diff
@@ -0,0 +1,154 @@
+import { useEffect, useRef, useState } from 'react'
+import styled from 'styled-components';
+import axios from 'axios';
+
+
+function BeerList() {
+  const [beerList, setBeerList] = useState([])
+  const [location, setLocation] = useState([])
+
+  const articleRef = useRef(null);
+
+  
+  // useEffect(() => {
+  //   getBeerList();
+  // }, []);
+
+
+  // 현재 위치의 위도와 경도 정보
+  const getLocation = (errHandler) => {
+    if ("geolocation" in navigator) {
+        return new Promise((resolve) => {
+        navigator.geolocation.getCurrentPosition(
+          (position) => {
+            const {
+              coords: { latitude: y, longitude: x },
+            } = position;
+            resolve({ x, y });
+          },
+          (e) => {
+            alert("HTTPS 연결을 확인해주세요.");
+            errHandler && errHandler();
+          }
+        );
+      });
+    }
+  
+    return { x: 126.8350976, y: 37.617664 };
+  };
+  
+  
+  
+    // 1 (using async-await)
+    async function getBeerList() {
+      const result = await getLocation();
+      //getBeerStore(result)
+      setLocation(result)
+    }
+    
+    
+    // 2 (using promise / then)
+    // getLocation().then(position => {
+    //    // console(position);
+    //    // getBeerStore(position)
+    // })
+    
+    
+    // 내 근처 맥주집 가져오기
+    async function getBeerStore() {
+      const {
+        data: { documents },
+      } = await axios.get(
+        "https://dapi.kakao.com/v2/local/search/keyword",
+        {
+          headers: {
+            Authorization: `KakaoAK ${ process.env.VITE_APP_KAKAO_AK }`,

```

Comment:
```suggestion
            Authorization: `KakaoAK ${ import.meta.env.VITE_APP_KAKAO_AK }`,
```

이게 뭔가 싶으시겠지만 vite에서 환경변수 불러올 때는 이렇게 불러와요!

Reply:
옷ㅎㅎ 감사합니다!!!! 다시 고쳐보겠습니다😊😊